.github/workflows/pico-build.yaml

```diff
@@ -36,8 +36,52 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  build-atomvmlib:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: "apt update"
+        run: sudo apt update
+
+      - name: "Install deps"
+        run: |
+          sudo apt install -y \
+              cmake doxygen gperf ninja-build gcc-arm-none-eabi \
+              libnewlib-arm-none-eabi libstdc++-arm-none-eabi-newlib \
+              erlang-base erlang-dev erlang-dialyzer erlang-eunit rebar3
+
+      - name: Build atomvmlib-pico.uf2/atomvmlib-pico2.uf2
+        shell: bash
+        run: |
+          set -euo pipefail
+          mkdir build
+          cd build
+          cmake .. -G Ninja
+          ninja
+
+      - name: Upload atomvmlib-pico artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: atomvmlib-pico.uf2
+          path: build/libs/atomvmlib-pico.uf2
```

Comment:
I would do build/libs/*.uf2, so we can benefit from JIT-precompiled uf2 (even if they don't fit on the Pico boards as such)